### PR TITLE
BCF-2555: unify job spec relayer access

### PR DESCRIPTION
**Pull request by krehermann**
Simplify access pattern for obtaining a relayer and chain from OCR2 job spec

### CODE REVIEW

File: core/services/job/models.go [hunk 1]
```diff
@@ -359,17 +359,14 @@ type OCR2OracleSpec struct {
 }
 
 func (s *OCR2OracleSpec) RelayIdentifier() (relay.Identifier, error) {
```
Comment (jmank88) [inline core/services/job/models.go:361]: WDYT about s/Identifier/ID? :shrug: 
```suggestion
RelayID uniquely identifies a relay.
func (s *OCR2OracleSpec) RelayID() (relay.Identifier, error) {
```

File: core/services/ocr2/delegate.go [hunk 2]
```diff
@@ -247,66 +248,62 @@ func (d *Delegate) OnDeleteJob(jb job.Job, q pg.Queryer) error {
...
-			return errors.Wrapf(err, "Failed to unregister filter %s", filter)
+		for _, filter := range filters {
+			d.lggr.Debugf("Unregistering %s filter", filter)
+			err = lp.UnregisterFilter(filter, pg.WithQueryer(q))
+			if err != nil {
+				return errors.Wrapf(err, "Failed to unregister filter %s", filter)
+			}
 		}
-	}
+	} // EVM specific cleanup
```
Comment (jmank88) [inline core/services/ocr2/delegate.go:306]: Would it make sense to group this evm-only block in a helper method? Maybe not if too many in-scope variables would have to be passed along :balance_scale:

File: core/services/job/models.go (hunk 1 above)
Comment (krehermann) [reply core/services/job/models.go]: done

File: core/services/ocr2/delegate.go (hunk 2 above)
Comment (krehermann) [reply core/services/ocr2/delegate.go]: yeah, thought about that too. done now

**Review by jmank88 — APPROVED** [review]

**Review by patrickhuie19 — APPROVED** [review]

**Pull request merged**